src/main/java/org/mongojack/internal/MongoAnnotationIntrospector.java

```diff
@@ -83,8 +83,7 @@ public Object findSerializer(Annotated am) {
     @Override
     public Object findDeserializer(Annotated am) {
         if (am.hasAnnotation(ObjectId.class)) {
-            return findObjectIdDeserializer(typeFactory.constructType(am
-                    .getGenericType()));
+            return findObjectIdDeserializer(typeFactory.constructType(am.getType()));
```

Comment:
Good question. The PR is poorly named. I have also upgrade jackson (needed for a module I plan to publish elsewhere plus fixes a bunch of jackson issues).

getGenericType has been deprecated in jackson 2.7, to be removed in 2.8, but also appears to have a subtle and likely unintentional behavior change. getType() does what mongojack needs.